pkg/cimfs/cim_writer_windows.go

```diff
@@ -62,6 +64,65 @@ func Create(imagePath string, oldFSName string, newFSName string) (_ *CimFsWrite
 	return &CimFsWriter{handle: handle, name: filepath.Join(imagePath, fsName)}, nil
 }
 
+func validateCreateCIMArgs(blockPath, oldName, newName string) error {
+	if blockPath == "" {
+		return fmt.Errorf("blockPath can not be empty: %w", os.ErrInvalid)
+	}
+	if oldName == "" && newName == "" {
+		return fmt.Errorf("both oldName & newName can not be empty: %w", os.ErrInvalid)
+	}
+	if oldName != "" && newName != "" {
+		return fmt.Errorf("both oldName & newName can not be provided: %w", os.ErrInvalid)
+	}
+	return nil
+}
+
+// Create creates a new blocked CIM or opens an existing one for writing. The CimFsWriter
+// returned can then be used to add/remove files to/from this CIM.  Exactly one of oldName
+// & newName must be non-empty string. If oldName is provided an existing block CIM is
+// opened for writing. If newName is provided a new block CIM is created and opened for
+// writing.
+func CreateBlockCIM(blockPath, oldName, newName string, blockType BlockCIMType) (_ *CimFsWriter, err error) {
```

Comment:
It seems like it would be simpler and easier to understand to just expose separate `CreateBlockCIM`/`OpenBlockCIM` functions, rather than taking old and new names.